data_structures/binary_search_tree.cpp

```diff
@@ -122,7 +155,33 @@ void Post(node *n) {
     }
 }
 
+void test_tree() {
+    node *root = new node;
+    root->val = 4;
+    root->left = NULL;
+    root->right = NULL;
+    // test Insert()
+    Insert(root, 2);
+    Insert(root, 1);
+    Insert(root, 3);
+    Insert(root, 6);
+    Insert(root, 5);
+    Insert(root, 7);
+    std::cout
```

Comment:
should perform self-check without manual intervention; something like below:
```cpp
for(int i = 1; i < 8; i++)
    assert(root[i] == i);
```

Reply:
Also, `2` is inserted first, so, shouldn't the expected output be 
`2,1,3,6,5,7`?

Reply:
> should perform self-check without manual intervention; something like below:
> 
> ```c++
> for(int i = 1; i < 8; i++)
>     assert(root[i] == i);
> ```

OK，I have use assert() instead. 
Many thanks!

Reply:
> Also, `2` is inserted first, so, is the output expected to be
> `2,1,3,6,5,7` or
> `1, 2, 3, 4, 5, 6 ,7`?
Here, i want to build a special tree.
When use Inorder traversal a binary search tree, the ouput is a sorted array:)